local/rest_api_gcbm/gcbm.py

```diff
@@ -0,0 +1,420 @@
+        self.json_paths.update(self.miscellaneous.setattr(file_path, payload))
+
+
+    # TODO: category should be an enum instead of a string to avoid any mistakes
+    def set_attributes(self, category: str, file_path: str, payload: dict):
+        base_path = os.path.basename(file_path)
+        if category == "disturbances":
+            self.set_disturbance_attributes(base_path, payload)
+        elif category == "classifiers":
+            self.set_classifier_attributes(base_path, payload)
+        elif category == "miscellaneous":
+            self.set_miscellaneous_attributes(base_path, payload)
+        else:
+            raise UserWarning(f"Expected a valid category name out of {self.supported_categories}, but got {category}")
+
+
+    @staticmethod
+    def safe_read_json(file_path):
+
+        # TODO: add read method for gcbm_config.cfg and logging.conf
+        if ".cfg" in file_path:
+            filename = os.path.join('input/test-run', "gcbm_config.cfg")
+            shutil.copy(file_path, filename)
+            return {}
+
+        if ".conf" in file_path:
+            filename = os.path.join('input/test-run', "logging.conf")
+            shutil.copy(file_path, filename)
+            return {}
+
+        # check JSON
+        if ".json" not in file_path:
+            raise UserWarning(f"Given path {file_path} not a valid json file")
+
+        # Make sure it's a file and not a directory
+        if not os.path.isfile(file_path):
+            raise UserWarning(
+                f"Got a directory {file_path} inside the config directory path, skipping it."
+            )
+        with open(file_path, "r") as json_file:
+            data = json.load(json_file)
+        return data
+
+
+if __name__ == "__main__":
+    sim = GCBMSimulation()
+    sim.add_file("tests/tiff/new_demo_run/disturbances_2011_moja.tiff")
+    # sim.set_disturbance_attributes("disturbances_2011_moja.tiff", {"year": 2011, "disturbance_type": "Wildfire", "transition": 1})
+    sim.set_attributes(category="disturbances", file_path="disturbances_2011_moja.tiff", payload={"year": 2011, "disturbance_type": "Wildfire", "transition": 1})
+
+    sim.add_file("tests/tiff/new_demo_run/disturbances_2012_moja.tiff")
+    sim.set_disturbance_attributes("disturbances_2012_moja.tiff",
+                                  {"year": 2012, "disturbance_type": "Wildfire", "transition": 1})
+
+
+    sim.add_file("tests/tiff/new_demo_run/disturbances_2013_moja.tiff")
+    sim.set_disturbance_attributes("disturbances_2013_moja.tiff",
+                                    {"year": 2013, "disturbance_type": "Mountain pine beetle — Very severe impact", "transition": 1})
+
+    # TODO: Check how to handle multiple attributes entries (L442-451 of `app.py:master`)
+    # sim.set_disturbance_attributes("disturbances_2013_moja.tiff",
+    #                                 {"year": 2013, "disturbance_type": "Wildfire", "transition": 1})
```

Comment:
This is important too - the shortest fix for adding multiple attributes might be to combine both into a single payload/

Reply:
I have taken a note of this, and will fix this TODO in the next commit. :)

Reply:
Done! :) Hope it looks good?